### PR TITLE
Added network direction in log

### LINKED ISSUE #46: New packet debug log format?
Can I implement a new debug log format?

The current debug format is a bit hard to read and requires a little bit of thinking
```
16:35:57 | Debug   | WorldClient | Received opcode SMSG_UPDATE_OBJECT (169).
16:35:57 | Debug   | WorldSocket | Sending opcode SMSG_UPDATE_OBJECT (10178).
17:35:58 | Debug   | WorldSocket | Received opcode CMSG_TALK_TO_GOSSIP (13458).
17:35:58 | Debug   | WorldClient | Sending opcode CMSG_TALK_TO_GOSSIP (379) with size 12.
17:35:58 | Debug   | WorldClient | Received opcode SMSG_GOSSIP_MESSAGE (381).
17:35:58 | Debug   | WorldSocket | Sending opcode SMSG_GOSSIP_MESSAGE (10904).
```
the new format would look like this:
```
16:35:57 | Debug   | C P<S | Received opcode SMSG_UPDATE_OBJECT (169).
16:35:57 | Debug   | C<P S | Sending opcode SMSG_UPDATE_OBJECT (10178).
17:35:58 | Debug   | C>P S | Received opcode CMSG_TALK_TO_GOSSIP (13458).
17:35:58 | Debug   | C P>S | Sending opcode CMSG_TALK_TO_GOSSIP (379) with size 12.
17:35:58 | Debug   | C P<S | Received opcode SMSG_GOSSIP_MESSAGE (381).
17:35:58 | Debug   | C<P S | Sending opcode SMSG_GOSSIP_MESSAGE (10904).
```
C = Client, P = HermesProxy, S = Server

The old class can stay in the log if you want.
```
16:35:57 | Debug   | WorldClient | C P<S | Received opcode SMSG_UPDATE_OBJECT (169).
```

### PR DESCRIPTION
**Pull request by 0blu**
Fix #46 

image

I don't know if this should be optional via config.

### CODE REVIEW

**Comment by 0blu** [comment]
Maybe for better debugging the TimeStamp should also not be optional and include milliseconds.

**Comment by 0blu** [comment]
Just so I can get a notice in advance; this is my pull request schedule for the future:
```
1. Display Char Count in Realmlist
(2. Remove singleton in auth)
3. Option in config to create an extra file which stores the last selected realm for each account
4. Fix "character rename required" at character selection

5. Remove some reflection with compiletime generators
6. Git hash when compiling. (For build message in log)

7. GitHub Action "Release Action"
8. Single Executable builds
```

File: HermesProxy/World/Server/WorldSocket.cs [hunk 1]
```diff
@@ -687,7 +687,7 @@ void HandleConnectToFailed(ConnectToFailed connectToFailed)
                     break;
                 case ConnectToSerial.WorldAttempt5:
                 {
-                    Log.Print(LogType.Error, "Failed to connect 5 times to world socket, aborting login");
+                    Log.PrintNet(LogType.Error, LogNetDir.S2P, "Failed to connect 5 times to world socket, aborting login");
```
Comment (ratkosrb) [inline HermesProxy/World/Server/WorldSocket.cs:690]: I think PrintNet only makes sense to be used for printing opcodes being sent/received, not other errors like that. And S2P which you use here is not even accurate, this error is about client failing to connect to instance socket of proxy.

Comment (0blu) [reply HermesProxy/World/Server/WorldSocket.cs]: Okay, I updated the commit but left the cases where it is actually network related (like sending or receiving packets).

**Pull request merged**